src/FixerConfiguration/FixerOption.php

```diff
@@ -0,0 +1,225 @@
+
+    /**
+     * @var bool
+     */
+    private $useDefault = false;
+
+    /**
+     * @var array|null
+     */
+    private $allowedTypes;
+
+    /**
+     * @var array|null
+     */
+    private $allowedValues;
+
+    /**
+     * @var \Closure|null
+     */
+    private $normalizer;
+
+    /**
+     * @param string $name
+     * @param string $description
+     */
+    public function __construct($name, $description)
+    {
+        $this->name = $name;
+        $this->description = $description;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return $this->description;
+    }
+
+    /**
+     * @param mixed $default
+     *
+     * @return $this
+     */
+    public function setDefault($default)
+    {
+        $this->default = $default;
+        $this->useDefault = true;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasDefault()
+    {
+        return $this->useDefault;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefault()
+    {
+        if (!$this->hasDefault()) {
+            throw new \LogicException('No default value defined.');
+        }
+
+        return $this->default;
+    }
+
+    /**
+     * @param string[]|string|null $allowedTypes
+     *
+     * @return $this
+     */
+    public function setAllowedTypes($allowedTypes)
+    {
+        if (null !== $allowedTypes && !is_array($allowedTypes)) {
+            $allowedTypes = array($allowedTypes);
+        }
+
+        $this->allowedTypes = $allowedTypes;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAllowedTypes()
+    {
+        return $this->allowedTypes;
+    }
+
+    /**
+     * @param string[]|string|null $allowedValues
+     *
+     * @return $this
+     */
+    public function setAllowedValues($allowedValues)
+    {
+        if (!is_array($allowedValues)) {
+            $allowedValues = array($allowedValues);
+        }
+
+        $this->allowedValues = $allowedValues;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAllowedValues()
+    {
+        return $this->allowedValues;
+    }
+
+    /**
+     * Sets the given option to only accept an array with a subset of the given values.
+     *
+     * @param array $allowedArrayValues
+     *
+     * @return $this
+     */
+    public function setAllowedValueIsSubsetOf(array $allowedArrayValues)
+    {
+        $option = $this->name;
+
+        return $this
+            ->setAllowedTypes('array')
+            ->setAllowedValues($this->unbind(function ($values) use ($option, $allowedArrayValues) {
+                foreach ($values as $value) {
+                    if (!in_array($value, $allowedArrayValues, true)) {
+                        throw new InvalidOptionsException(sprintf(
```

Comment:
Would work too, but the error message would be something like _Option "foo" with value "array" is not valid_, which is less helpful that the error message here.